test/Octonion.jl

```diff
@@ -85,57 +164,235 @@ using Test
-            test_group(q1, q2, q3, *, one(q), inv)
+            # test all group properties but associativity
+            test_neutral(q1, one(q1), *)
+            test_inverse(q1, one(q1), *, inv)
             test_multiplicative(q1, q2, *, norm)
 
             # complex embedding
-            test_multiplicative(c1, c2, *, Quaternion)
-            test_multiplicative(c1, c2, +, Quaternion)
+            test_multiplicative(c1, c2, *, Octonion)
+            test_multiplicative(c1, c2, +, Octonion)
         end
     end
 
-    @testset "isreal" begin end
-
-    @testset "iszero" begin end
-
-    @testset "isone" begin end
+    @testset "isreal" begin
+        @test isreal(octo(1))
+        @test !isreal(octo(1, 1, 0, 0, 0, 0, 0, 0))
+        @test !isreal(octo(1, 0, 1, 0, 0, 0, 0, 0))
+        @test !isreal(octo(1, 0, 0, 1, 0, 0, 0, 0))
+        @test !isreal(octo(1, 0, 0, 0, 1, 0, 0, 0))
+        @test !isreal(octo(1, 0, 0, 0, 0, 1, 0, 0))
+        @test !isreal(octo(1, 0, 0, 0, 0, 0, 1, 0))
+        @test !isreal(octo(1, 0, 0, 0, 0, 0, 0, 1))
+    end
 
-    @testset "isfinite" begin end
+    @testset "iszero" begin
+        @test iszero(octo(0))
+        @test !iszero(octo(1))
+        @test !iszero(octo(0, 1, 0, 0, 0, 0, 0, 0))
+        @test !iszero(octo(0, 0, 1, 0, 0, 0, 0, 0))
+        @test !iszero(octo(0, 0, 0, 1, 0, 0, 0, 0))
+        @test !iszero(octo(0, 0, 0, 0, 1, 0, 0, 0))
+        @test !iszero(octo(0, 0, 0, 0, 0, 1, 0, 0))
+        @test !iszero(octo(0, 0, 0, 0, 0, 0, 1, 0))
+        @test !iszero(octo(0, 0, 0, 0, 0, 0, 0, 1))
+        @test !iszero(octo(0, 0, 0, 0, 0, 0, 0, 0, true))
+    end
 
-    @testset "isinf" begin end
+    @testset "isone" begin
+        @test isone(octo(1))
+        @test !isone(octo(-1))
+        @test !isone(octo(0, 1, 0, 0, 0, 0, 0, 0))
+        @test !isone(octo(1, 1, 0, 0, 0, 0, 0, 0))
+        @test !isone(octo(1, 0, 1, 0, 0, 0, 0, 0))
+        @test !isone(octo(1, 0, 0, 1, 0, 0, 0, 0))
+        @test !isone(octo(1, 0, 0, 0, 1, 0, 0, 0))
+        @test !isone(octo(1, 0, 0, 0, 0, 1, 0, 0))
+        @test !isone(octo(1, 0, 0, 0, 0, 0, 1, 0))
+        @test !isone(octo(1, 0, 0, 0, 0, 0, 0, 1))
+    end
 
-    @testset "isnan" begin end
+    @testset "isfinite" begin
+        @test isfinite(octo(1:8...))
+        for inf in (Inf, -Inf)
+            @test !isfinite(octo(inf, 0, 0, 0, 0, 0, 0, 0))
+            @test !isfinite(octo(0, inf, 0, 0, 0, 0, 0, 0))
+            @test !isfinite(octo(0, 0, inf, 0, 0, 0, 0, 0))
+            @test !isfinite(octo(0, 0, 0, inf, 0, 0, 0, 0))
+            @test !isfinite(octo(0, 0, 0, 0, inf, 0, 0, 0))
+            @test !isfinite(octo(0, 0, 0, 0, 0, inf, 0, 0))
+            @test !isfinite(octo(0, 0, 0, 0, 0, 0, inf, 0))
+            @test !isfinite(octo(0, 0, 0, 0, 0, 0, 0, inf))
+            @test isfinite(octo(fill(inf, 8)..., true))
+        end
+    end
 
-    @testset "+" begin end
+    @testset "isinf" begin
+        @test !isinf(octo(1:8...))
```

Comment:
```suggestion
        @test !isinf(octo(1, 2, 3, 4, 5, 6, 7, 8))
        @test !isinf(octo(1, 2, 3, 4, 5, 6, 7, NaN))
```
The change from `1:8` to `1, 2, 3, 4, 5, 6, 7, 8` is to conform to `isnan` test.